include/range/v3/view_interface.hpp

```diff
@@ -297,6 +297,32 @@ namespace ranges
             {
                 return Slice{}(detail::move(derived()), offs.from, offs.to);
             }
+            /// Returns a reference to the element at specified location pos, with bounds checking.
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(Same<D, Derived>() && RandomAccessRange<D>() && SizedRange<D>())>
+            auto at(range_difference_type_t<D> n) ->
+                decltype(std::declval<D &>().begin()[n])
+            {
+                using size_type = range_size_type_t<Derived>;
+                if (n < 0 || size_type(n) >= ranges::size(derived()))
+                {
+                    throw std::out_of_range("view_interface::at");
+                }
+                return derived().begin()[n];
```

Comment:
Note that we're applying the bracket operator on a random access iterator. As a result, it _should_ accept a difference type rather than a size type. Consider the following:

```cpp
int main() {
    std::vector<int> v{1,2,3,4};
    assert( v[2] == v.end()[-2] );
 }
```
Consider what whould happen here if the bracket operator accepted a size type rather than a difference type.